sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -336,12 +336,7 @@ class Dataset[T] private[sql](
   def mapPartitions[U : Encoder](func: Iterator[T] => Iterator[U]): Dataset[U] = {
     new Dataset[U](
       sqlContext,
-      MapPartitions[T, U](
-        func,
-        resolvedTEncoder,
-        encoderFor[U],
-        encoderFor[U].schema.toAttributes,
-        logicalPlan))
+      MapPartitions[T, U](func, logicalPlan))
```

Comment:
This is different from the previous one, we only pass the type parameter `T` to `MapPartitions` and build a new encoder there which is unresolved, while before this PR we pass a `resolvedTEncoder`. Do we break the life cycle of encoder in this PR?